### ISSUE TITLE
Duplicate carriers when using PostNL custom contracts

### ISSUE THREAD

**Issue by mariuszsienkiewicz** [myparcelnl/pdk#284]
@EdieLemoine: When fetching carriers from the MyParcel API, when PostNL custom contract carrier is enabled, multiple postnl carriers are returned. This causes problems all over the place. We need to correct this manually before saving these carriers in the PDK.

### Version check

- [X] My version of the plugin is v4.0.0-beta.0 or above.

### PrestaShop version

8.1.5

### PHP version

8.1.28

### What went wrong?

I have tried to test the myparcelnl on stock PrestaShop 8.1.5 installation.

It seems that there is an issue when trying to save the API key for the first time (and after deleting the API key and setting it up again the issue occurs too).

Error log is below.

From what I can understand, the issue is that PostNL Carrier is duplicated (I don't know why) and the PostNL (Custom contract) use the same `name`. `CarrierCollection` which is used in the process of setting up new carriers contain seven `MyParcelNL\Pdk\Carrier\Model\Carrier` objects and three of them have the same name:

```
[0] => MyParcelNL\Pdk\Carrier\Model\Carrier Object
    (
    [cloned:protected] => 
    [attributes:protected] => Array
            (
                [externalIdentifier] => 
                [id] => 1
                [name] => postnl
                [human] => PostNL
                [contractId] => 
                [enabled] => 1
                [isDefault] => 1
                [label] => 
                [optional] => 0
                [primary] => 1
                [type] => main
```

```
[1] => MyParcelNL\Pdk\Carrier\Model\Carrier Object
    (
    [cloned:protected] => 
    [attributes:protected] => Array
            (
                [externalIdentifier] => 
                [id] => 1
                [name] => postnl
                [human] => PostNL
                [contractId] => 
                [enabled] => 1
                [isDefault] => 1
                [label] => 
                [optional] => 0
                [primary] => 1
                [type] => main
```

```
[2] => MyParcelNL\Pdk\Carrier\Model\Carrier Object 
    (
    [cloned:protected] => 
        [attributes:protected] => Array
            (
                [externalIdentifier] => 
                [id] => 1
                [name] => postnl
                [human] => PostNL
                [contractId] => <redacted>
                [enabled] => 1
                [isDefault] => 1
                [label] => 
                [optional] => 0
                [primary] => 1
                [type] => custom
```

There is one more bug (duplicated PostNL) in the configuration page too (screenshot below) that is probably related to this issue.

### Reproduction steps

1. Install the plugin on stock PrestaShop 8.1.5
2. Go to the configuration page
3. Save the API key - error
4. Refresh
+ Configuration page now shows in the Carriers tab two PostNL tabs.

### Relevant log output

```shell
accountUpdate ERROR handler ApiException: An exception occurred while executing 'INSERT INTO ps_myparcelnl_carrier_mapping (carrier_id, myparcel_carrier, date_upd, date_add) VALUES (?, ?, ?, ?)' with params [6, "postnl", "2024-04-24 13:33:08", "2024-04-24 13:33:08"]: SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry 'postnl' for key 'myparcel_carrier'
    rn https://domain.com/modules/myparcelnl/views/js/backend/admin/dist/index.iife.js:15
    doRequest https://domain.com/modules/myparcelnl/views/js/backend/admin/dist/index.iife.js:21
index.iife.js:21:1018
    log https://domain.com/modules/myparcelnl/views/js/backend/admin/dist/index.iife.js:21
    error https://domain.com/modules/myparcelnl/views/js/backend/admin/dist/index.iife.js:21
    Xy https://domain.com/modules/myparcelnl/views/js/backend/admin/dist/index.iife.js:21
```

### Additional context

![image](https://github.com/myparcelnl/prestashop/assets/62615548/c460403b-c0e1-4b4b-9459-6dc1a3fad390)

**Comment by maikelwever** [comment]
I have the exact same problem, but I don't have any custom contracts with PostNL. It seems to be related to the duplicate entries for regular non-custom-contract PostNL.

```An exception occurred while executing 'INSERT INTO ps_myparcelnl_carrier_mapping (carrier_id, myparcel_carrier, date_upd, date_add) VALUES (?, ?, ?, ?)' with params [36, "postnl", "2024-07-05 16:42:59", "2024-07-05 16:42:59"]:SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry 'postnl' for key 'myparcel_carrier'```

PrestaShop version: 8.1.7 
PHP version: 8.3.8 (also tried with 8.1.28, same result)
MyParcel Module version: myparcelnl-prestashop-4.0.0-beta.3.zip (reports as v4.0.0-be)

The plugin also creates duplicate entries for PostNL in the PrestaShop carriers list:

![image](https://github.com/myparcelnl/pdk/assets/1009019/be404f81-a5aa-4184-b77a-a4738d1c8293)

**Comment by EdieLemoine** [comment]
This will be fixed when myparcelnl/pdk#287 is live, which will then be included in the next release of the module.

**Comment by mariuszsienkiewicz** [comment]
@EdieLemoine The duplicated carrier issue is still present in version 4.2.0 of the MyParcel module. Unfortunately, the problem persists, and we're unable to make any adjustments to the PostNL carrier configuration. :(

It feels like version 4.2.0 is still far from being "stable." There are still several unresolved issues with this module.

By the way, is it normal for the MyParcel module to "migrate" the labels with every upgrade? We have over 100,000 orders with existing labels, and we thought this process would only happen once—during the upgrade to version 4.0.0. However, it seems to occur with every upgrade. It takes several hours, which, as you can imagine, is far from convenient.
